### ISSUE TITLE
improving search results

### ISSUE THREAD

**Issue by br4nnigan** [openstreetmap/Nominatim#889]
I think one of the biggest issues of OSM are the search results.

For instance the term "14129 Berlin" will only show the zipcode associated with Berlin Germany at 9th place, at the end of the list and give me a tiny street named "Berlin" in Chile as best hit.

Therefore cities should be favoured over streets, countries over cities etc.

But more importantly: why does OSM not use geolocation to rate the search results? Geolocation can heavily improve search results. If I'm in Berlin and get hits for Berlin, then those have to have the highest priority. Distance should play a major role in rating the search results. I can see privacy concerns here, but as the Browser has to ask for permission anyway I don't see the issue.

**Comment by lonvia** [comment]
Nominatim does favour cities over streets and countries over cities. Nominatim also does take geolocation into account. So I guess we are good here. Closing.

**Comment by br4nnigan** [comment]
I see. The problem seems to be that using a zip code like I did prevents Nominatim from identifying the city. Should we create a seperate ticket for that? @lonvia 

The result I got in 9th place from the city is actually an artwork coincidently existing.

Also it seems there is a problem with providing my geolocation in my browser, since google too cannot use it despite me giving permission (local computer, not phone).
